src/draggable.ts

```diff
@@ -42,27 +57,88 @@ export class Draggable extends EventTarget {
       options: DraggableOptions = {}) {
     super();
 
+    this.over = {tag: 'NonTarget'};
+
     this.options = applyDefaults(options, {moveX: true, moveY: true});
+    this.$targets = this.options.$targets;
     this.setDimensions($parent);
 
-    let startPosn: Point;
     slide($el, {
       start: () => {
         if (this.disabled) return;
-        startPosn = this.position;
+        this.startPos = this.position;
         this.trigger('start');
         $html.addClass('grabbing');
       },
       move: (posn, start) => {
         if (this.disabled) return;
-        this.setPosition(startPosn!.x + posn.x - start.x,
-                         startPosn!.y + posn.y - start.y);
-        this.trigger('drag', this.position);
+        this.setPosition(this.startPos.x + posn.x - start.x,
+            this.startPos.y + posn.y - start.y);
+        this.trigger('drag', {elemPos: this.position, pointerPos: posn});
+
+        let overTarget = false;
+        const prevTarget = this.over.tag == 'NonTarget' ? undefined : this.over.$el;
+        if (this.$targets != undefined) {
+          for (const $target of this.$targets) {
+            if ($target.boundsRect.contains(posn)) {
+              overTarget = true;
+              if (prevTarget != $target) {
+                /**
+                 * Fires when the pointer has entered the bounds of a 'target' element while dragging
+                 *
+                 * @event Draggable#enter-target
+                 * @type {ElementView} $target - The element whose bounds the pointer has entered
+                 */
+                this.trigger('enter-target', $target);
```

Comment:
make the event callback an object (e.g. `this.trigger('enter-target', {$target})` in case we want to add other properties in the future